### PR TITLE
Synchronously save chunks on server shutdown

### LINKED ISSUE #4884: Wolfes sometimes loose color on collar on server restart
Client version: 1.11.1
Server OS: Windows
Cuberite Commit id: 53549a1a4ce320c1e3f449165c290c74d3de54d4

### Expected behavior
Collars can be colored and should keep their color on server restart

### Actual behavior
Color loss on restart

### Steps to reproduce the behavior
- tame wolf
- color collar
- stop server
- restart

### PR DESCRIPTION
**Pull request by bearbin**
Fixes #4884

### CODE REVIEW

**Comment by peterbell10** [comment]
Shouldn't we make sure all queued tasks get executed and not just this special case?

**Comment by tigerw** [comment]
Queued tasks also have a (potentially long) tick delay.

**Comment by bearbin** [comment]
Potentially we could add a `RunAllQueuedTasksNowAndWaitForThemToFinish` function on cWorld, and then use that instead, but we would still need a special case for saving, because we would want the saving to run after all of the other queued tasks (wouldn't we?).

File: Server/Plugins/APIDump/APIDesc.lua [hunk 1]
```diff
@@ -11653,6 +11653,10 @@ a_Player:OpenWindow(Window);
 				{
 					Notes = "Saves all the chunks in all the worlds. Note that the saving is queued on each world's tick thread and this functions returns before the chunks are actually saved.",
 				},
+				SaveAllChunksNow =
```
Comment (tigerw) [inline Server/Plugins/APIDump/APIDesc.lua:11656]: I would hold off exporting this, it could introduce deadlocks if plugins try to use it in a world thread

**Comment by bearbin** [comment]
I was pushing that change just as you were writing the comment!

**Comment by peterbell10** [comment]
Maybe the queue could be completed during world shutdown? We need to be sure nothing will be queued up after the queue is emptied.

**Comment by tigerw** [comment]
> I was pushing that change just as you were writing the comment!

Synchronously ~~save chunks~~ post review comments on ~~server shutdown~~ commit push.

File: src/Server.cpp [hunk 2]
```diff
@@ -648,7 +648,8 @@ void cServer::Shutdown(void)
 	// Notify the tick thread and wait for it to terminate:
 	m_TickThread.Stop();
 
-	cRoot::Get()->SaveAllChunks();
+	// Save all chunks in all worlds, wait for chunks to be saved:
```
Comment (tigerw) [inline src/Server.cpp:651]: I just looked at cWorld::SaveAllChunks and apparently it only adds it to the world storage queue, it doesn't wait for it to empty. However cRoot waits for the thread to finish to it's still ok, just the comment might be wrong. Could you confirm?

Comment (bearbin) [reply src/Server.cpp:651]: Your comment is completely correct, I'll update my comment.

**Review by tigerw — APPROVED** [review]

**Pull request merged**